main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala

```diff
@@ -172,44 +173,47 @@ object BuildServerProtocol {
       )
     )
 
-  private def bspBuildTargetSetting: Def.Initialize[BuildTarget] = Def.settingDyn {
+  private def bspBuildTargetTask: Def.Initialize[Task[BuildTarget]] = Def.taskDyn {
```

Comment:
It's too bad build target had to become a task. I tried to keep that a synchronous thing in #5443

Reply:
It has become  a task because sbt needs to retrieve the scala libraries in order to fill the `jars` field of `ScalaBuildTarget`. In #5443 this [field was left empty](https://github.com/eed3si9n/sbt/blob/c31b42085ad52f98ea7b6cb8b33881688e3b1880/main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala#L150) which is ok for some client (Metals for instance) but maybe not for all. 

The situation is the same with `build/scalacOptions` because sbt must retrieve the external dependencies to build the classpath. If we fix this for `bspBuildTarget` we should fix it also for `scalacOptions`. How to? Maybe we can play with the cache to preserve synchronicity as much as possible but I have no idea if it is feasible or viable in the long term.

I would rather stay like that for now.

Reply:
Yea. That was done intentionally so we can return build target as soon as the build is loaded. Maybe we can ask the BSP maintainers for the clarification on this? Note that it's a `String[]`, not `Uri[]`, so I don't know if this is intended to return the full path to the JAR files.

For import process, I think makes sense to me that the sbt's internal idea of the build targets remain constant unless someone calls `reload`. I'd guess that the more we can reduce the tasks, the general experience of integrating with BSP would feel smoother.

Classpath JARs would be retrieved as task in `buildTarget/scalacOptions` *afterwards*.

Reply:
> Classpath JARs would be retrieved as task in buildTarget/scalacOptions afterwards.

I know from @jastice that the jars field is used in the IntelliJScala plugin because it contains the compiler jars, which are not in the classpath. So the jars field is absolutely necessary.

I really think that `workspace/buildTarget` and `buildTarget/scalacOptions` are complementary and that a client need both to be answered for the project to be imported. The best would be to have both be settings. Until then, it is not a big deal if the `workspace/buildTarget` is a task.

Reply:
We can split this as a new issue and discuss later/further with Justin. This is not a blocker for this PR.